onnxruntime/contrib_ops/cpu/bert/multihead_attention.cc

```diff
@@ -106,34 +125,54 @@ Status MultiHeadAttention<T>::Compute(OpKernelContext* context) const {
   const int k_bias_offset = qk_hidden_size;
   const int v_bias_offset = 2 * qk_hidden_size;
 
-  // If optional outputs aren't needed, present_k and present_v will be null
-  std::vector<int64_t> present_k_shape({static_cast<int64_t>(batch_size),
-                                        static_cast<int64_t>(num_heads_),
-                                        static_cast<int64_t>(total_kv_sequence_length),
-                                        static_cast<int64_t>(qk_head_size)});
-  std::vector<int64_t> present_v_shape({static_cast<int64_t>(batch_size),
+  // If optional outputs aren't needed, present_key, present_value, and output_qk will be null
+  std::vector<int64_t> present_key_shape({static_cast<int64_t>(batch_size),
+                                          static_cast<int64_t>(num_heads_),
+                                          static_cast<int64_t>(parameters.max_sequence_length),
```

Comment:
I understand that it is to support shared buffer for past and present.
previously, this dim is total sequence length (when no shared buffer). Is this change compatible with that?